core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/benchmark/DatatypeBenchmarkEmpty.java

```diff
@@ -78,77 +76,84 @@ public void setUp() throws InterruptedException {
+					connection.commit();
+				}
+			}
 
+			repository.shutDown();
+		}
 	}
 
 	@Benchmark
 	public void noShacl() {
+		try (Utils.TemporaryFolder temporaryFolder = Utils.newTemporaryFolder()) {
 
-		SailRepository repository = new SailRepository(new TestNotifyingSail(new MemoryStore()));
+			SailRepository repository = new SailRepository(
+					Utils.getTestNotifyingSailNativeStore(temporaryFolder));
 
-		repository.init();
+			repository.init();
 
-		try (SailRepositoryConnection connection = repository.getConnection()) {
-			connection.begin(IsolationLevels.SNAPSHOT);
-			connection.commit();
-		}
-		try (SailRepositoryConnection connection = repository.getConnection()) {
-			for (List<Statement> statements : allStatements) {
+			try (SailRepositoryConnection connection = repository.getConnection()) {
 				connection.begin(IsolationLevels.SNAPSHOT);
-				connection.add(statements);
 				connection.commit();
 			}
-		}
+			try (SailRepositoryConnection connection = repository.getConnection()) {
+				for (List<Statement> statements : allStatements) {
+					connection.begin(IsolationLevels.SNAPSHOT);
+					connection.add(statements);
+					connection.commit();
+				}
+			}
+			repository.shutDown();
 
-//		repository.shutDown();
+		}
 
 	}
 
 	@Benchmark
 	public void sparqlInsteadOfShacl() {
+		try (Utils.TemporaryFolder temporaryFolder = Utils.newTemporaryFolder()) {
 
-		SailRepository repository = new SailRepository(new MemoryStore());
+			SailRepository repository = new SailRepository(
+					Utils.getTestNotifyingSailNativeStore(temporaryFolder));
 
-		repository.init();
+			repository.init();
 
-		try (SailRepositoryConnection connection = repository.getConnection()) {
-			connection.begin(IsolationLevels.SNAPSHOT);
-			connection.commit();
-		}
-		try (SailRepositoryConnection connection = repository.getConnection()) {
-			for (List<Statement> statements : allStatements) {
+			try (SailRepositoryConnection connection = repository.getConnection()) {
 				connection.begin(IsolationLevels.SNAPSHOT);
-				connection.add(statements);
-				try (Stream<BindingSet> stream = connection
-						.prepareTupleQuery("select * where {?a a <" + RDFS.RESOURCE + ">; <" + FOAF.AGE
-								+ "> ?age. FILTER(datatype(?age) != <http://www.w3.org/2001/XMLSchema#int>)}")
-						.evaluate()
-						.stream()) {
-					stream.forEach(System.out::println);
-				}
 				connection.commit();
 			}
-		}
-
-//		repository.shutDown();
+			try (SailRepositoryConnection connection = repository.getConnection()) {
+				for (List<Statement> statements : allStatements) {
+					connection.begin(IsolationLevels.SNAPSHOT);
+					connection.add(statements);
+					try (Stream<BindingSet> stream = connection
+							.prepareTupleQuery("select * where {?a a <" + RDFS.RESOURCE + ">; <" + FOAF.AGE
+									+ "> ?age. FILTER(datatype(?age) != <http://www.w3.org/2001/XMLSchema#int>)}")
+							.evaluate()
+							.stream()) {
+						stream.forEach(System.out::println);
```

Comment:
Won't this generate rather a lot of noise on the console?

Reply:
Didn't catch that one.